### DISCUSSION TITLE
Milestone 1: Pokemon Premier League

### DISCUSSION THREAD

**Discussion by arpitbbhayani** [DiceDB/dice#46] [Ideas]
Milestone 1 for Dice DB would be to power **Pokemon Premier League**

We should be able to power the attached dashboard through Dice DB.

![Pokemon Premier League(2)](https://user-images.githubusercontent.com/4745789/196925700-d74f0ca9-327c-4130-ac54-940f0c30f222.png)

Datasets:
- https://www.kaggle.com/datasets/abcsds/pokemon
- https://www.kaggle.com/datasets/vishalsubbiah/pokemon-images-and-types

Design Document

- https://github.com/DiceDB/dice/wiki/Milestone-1:-Pokemon-Premier-League

Constraints:

 - storage engine will be in-memory
 - single-threaded execution loop
 - we may have commands that use multi-threaded scatter-gather for their execution

Let's use this discussion thread to see the kind of features we want to converge on.

**Comment by golu360** [comment]
Command to fetch min-max of a given key or subkey.

Say 

"Pokemon with max defense" (FINDMAX *.def) (where *=All pokemon and def=Defense stat).

Same goes for the Minimum Stat.
